mesonbuild/interpreter/kwargs.py

```diff
@@ -343,14 +343,16 @@ class _SharedLibMixin(TypedDict):
     darwin_versions: T.Optional[T.Tuple[str, str]]
     soversion: T.Optional[str]
     version: T.Optional[str]
+    vs_module_defs: T.Optional[T.Union[str, File, build.CustomTarget]]
```

Comment:
Do you want to use mesonlib.StrOrFile? I guess it is more characters to type at the end of the say.